dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -517,18 +444,25 @@ def find_counterfactuals(self, query_instance, desired_range, desired_class, fea
         else:
             self.final_cfs_df_sparse = None
 
-        # to display the values with the same precision as the original data
-        precisions = self.data_interface.get_decimal_precisions()
-        for ix, feature in enumerate(self.data_interface.continuous_feature_names):
-            self.final_cfs_df[feature] = self.final_cfs_df[feature].astype(float).round(precisions[ix])
-            self.final_cfs_df_sparse[feature] = self.final_cfs_df_sparse[feature].astype(float).round(precisions[ix])
+        if self.final_cfs_df is not None:
```

Comment:
precision-based output should be a part of explainer_base and common to all methods.